rest/admin_api_test.go

```diff
@@ -29,6 +29,60 @@ import (
 	"github.com/couchbaselabs/go.assert"
 )
 
+
+// Reproduces #3048 Panic when attempting to make invalid update to a conflicting document
+func TestNoPanicInvalidUpdate(t *testing.T) {
+
+	var rt RestTester
+	defer rt.Close()
+
+	docId := "conflictTest"
+
+	// Create doc
+	response := rt.SendAdminRequest("PUT", fmt.Sprintf("/db/%s", docId), `{"value":"initial"}`)
+	response.DumpBody()
+
+	// Discover revision ID
+	// TODO: Is there any way to leverage db.document here to avoid the type assertions?
```

Comment:
We define a temporary struct for this kind of scenario in a lot of tests, but really this suggests to me that the schema for SG responses should be defined in our code somewhere, even if it's more economical to build it as a simple Body in places like handlePutDoc.

Reply:
Yeah, ok I'll leave as-is and update the TODO